File: flybanner/src/main/java/com/xia/flybanner/constant/PageType.java
package com.xia.flybanner.constant;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.IntDef;

/**
 * @author weixia
 * @date 2019/4/23.
 */
public final class PageType {
    public static final int TYPE_NORMAL = 0;
    public static final int TYPE_GUIDE = 1;

    @IntDef({TYPE_NORMAL, TYPE_GUIDE})
    @Retention(RetentionPolicy.SOURCE)
    public @interface Type {
    }
}

File: flybanner/src/main/java/com/xia/flybanner/constant/PageType.kt
package com.xia.flybanner.constant

import androidx.annotation.IntDef

/**
 * @author weixia
 * @date 2019/4/23.
 */
object PageType {
    const val TYPE_NORMAL = 0
    const val TYPE_GUIDE = 1

    @IntDef(TYPE_NORMAL, TYPE_GUIDE)
    @Retention(AnnotationRetention.SOURCE)
    annotation class Type
}
